### PR TITLE
Integrate RBCMs onto development branch

### PR DESCRIPTION
**Pull request by stevetorr**
Under construction. RBCMs have been excised from the `xyz_loop_in_kernel` branch so that they can be used by a wider audience to take advantage of faster hyperparameter training.

- [x] Get RBCMs working alone.
- [x] Complete full unit tests & docstrings for RBCMs.

PR Review:

- @nw13slx  ensure that no changes have compromised the function of the RBCMs.

### CODE REVIEW

File: flare/gp_algebra.py [hunk 1]
```diff
@@ -1719,6 +1819,78 @@ def get_like_from_mats(ky_mat, l_mat, alpha, name):
...
+        l_mat = np.linalg.cholesky(ky_mat)
+    except np.linalg.LinAlgError:
+        number_of_hyps = len(hyps)
+        return -1e8, np.zeros(number_of_hyps)
+
+    labels = _global_training_labels[name]
+
+    alpha = np.matmul(ky_mat_inv, labels)
+    alpha_mat = np.matmul(alpha.reshape(-1, 1), alpha.reshape(1, -1))
+    like_mat = alpha_mat - ky_mat_inv
```
Comment (nw13slx) [inline flare/gp_algebra.py:1877]: I remember writing this function, but don't remember why I didn't use get_like_from_mat for the following lines... maybe double check whether the codes are repeated from line 1878 to the end of the function

File: flare/rbcm.py [hunk 2]
```diff
@@ -0,0 +1,1267 @@
...
+    Splits up prediction and hyperparameter optimzation over a 'comittee'
+    of experts, which gets around the unfavorable nonlinear scaling of
+    preidction and training time with training set size. Particularly for
+    optimizing hyperparameters (typically the most
+    time-consuming individual step of adapting GPs to a new system).
+
+    opt_algorithm GP kwarg supports "dual annealing" and "differential evolution"
+    for the RBCMs.
+
+    See https://arxiv.org/pdf/1806.00720.pdf for a theory background.
```
Comment (nw13slx) [inline flare/rbcm.py:58]: please cite the original paper instead: Deisenroth, M.P., and Ng, J.W. (2015). Distributed Gaussian Processes. 10.

Because we didn't use GRBCM proposed in the newer paper.

File: flare/rbcm.py [hunk 3]
```diff
@@ -0,0 +1,1267 @@
...
+
+        return gp_model
+
+    def __str__(self):
+        """String representation of the GP model."""
+
+        thestr = "GaussianProcess Object\n"
+        thestr += f"Number of cpu cores: {self.n_cpus}\n"
+        thestr += f"Kernel: {self.kernels}\n"
+        thestr += f"Training points: {len(self.training_data)}\n"
```
Comment (nw13slx) [inline flare/rbcm.py:1077]: maybe print the number of experts as well?

**Review by nw13slx — CHANGES_REQUESTED** [review]
Thanks for merging the RBCM implementation with the lastest development branch. The unit tests look great! Please change the reference, then it's ready to merge.

File: flare/gp_algebra.py (hunk 1 above)
Comment (stevetorr) [reply flare/gp_algebra.py:1877]: Good catch, and I fixed the repeated code. Looks like alpha mat and like mat aren't used.
